protocol/rpcprovider/rewardserver/rewards_store.go

```diff
@@ -0,0 +1,169 @@
+	}
+
+	buf, err := json.Marshal(re)
+	if err != nil {
+		return false, utils.LavaFormatError("failed to encode proof: %s", err)
+	}
+
+	rs.db.Save(ctx, key, buf)
+
+	return true, nil
+}
+
+func (rs *RewardStore) FindOne(
+	ctx context.Context,
+	epoch uint64,
+	consumerAddr string,
+	consumerKey string,
+	sessionId uint64,
+) (*pairingtypes.RelaySession, error) {
+	key := assembleKey(epoch, consumerAddr, sessionId, consumerKey)
+
+	rawReward, err := rs.db.FindOne(ctx, key)
+	if err != nil {
+		return nil, utils.LavaFormatDebug("reward not found")
+	}
+
+	var re RewardEntity
+	err = json.Unmarshal(rawReward, &re)
+	if err != nil {
+		return nil, utils.LavaFormatError("failed to decode proof: %s", err)
+	}
+
+	return re.Proof, nil
+}
+
+func (rs *RewardStore) FindAll(ctx context.Context) (map[uint64]*EpochRewards, error) {
+	rawRewards, err := rs.db.FindAll(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	result := make(map[uint64]*EpochRewards)
+	for key, rewards := range rawRewards {
+		epoch, consumerAddr, sessionId, consumerKey := keyParts(key)
+
+		re := RewardEntity{}
+		err := json.Unmarshal(rewards, &re)
+		if err != nil {
+			utils.LavaFormatError("failed to decode proof: %s", err)
+			continue
+		}
+
+		epochRewards, ok := result[epoch]
+		if !ok {
+			proofs := map[uint64]*pairingtypes.RelaySession{sessionId: re.Proof}
+			consumerRewards := map[string]*ConsumerRewards{consumerKey: &ConsumerRewards{epoch: epoch, consumer: consumerAddr, proofs: proofs}}
+			result[epoch] = &EpochRewards{epoch: epoch, consumerRewards: consumerRewards}
+			continue
+		}
+
+		consumerRewards, ok := epochRewards.consumerRewards[consumerKey]
+		if !ok {
+			proofs := map[uint64]*pairingtypes.RelaySession{sessionId: re.Proof}
+			epochRewards.consumerRewards[consumerKey] = &ConsumerRewards{epoch: epoch, consumer: consumerAddr, proofs: proofs}
+			continue
+		}
+
+		_, ok = consumerRewards.proofs[sessionId]
+		if !ok {
+			consumerRewards.proofs[sessionId] = re.Proof
+			continue
+		}
+	}
+	return result, nil
+}
+
+func (rs *RewardStore) DeleteClaimedRewards(ctx context.Context, claimedRewards []*pairingtypes.RelaySession) error {
+	var deletedPrefixes []string
+	for _, claimedReward := range claimedRewards {
+		consumer, err := sigs.ExtractSignerAddress(claimedReward)
+		if err != nil {
+			utils.LavaFormatError("failed to extract consumer address: %s", err)
+			continue
+		}
+
+		prefix := assembleKey(uint64(claimedReward.Epoch), consumer.String(), claimedReward.SessionId, "")
+		if slices.Contains(deletedPrefixes, prefix) {
+			continue
+		}
+
+		err = rs.db.DeletePrefix(ctx, prefix)
+		if err != nil {
+			utils.LavaFormatError("failed to delete rewards: %s", err)
+			continue
+		}
+
+		deletedPrefixes = append(deletedPrefixes, prefix)
+	}
+
+	return nil
+}
+
+func NewRewardStore(db DB) *RewardStore {
+	return &RewardStore{
+		db: db,
+	}
+}
+
+func assembleKey(epoch uint64, consumerAddr string, sessionId uint64, consumerKey string) string {
```

Comment:
Because we no longer need to disassmble keys (see previous comment) we now don't need a seperator and can simplify this method a bit 

` return strconv.FormatUint(epoch, 10) + consumerAddr + strconv.FormatUint(sessionId, 10) + consumerKey`

Reply:
The reason for the separator is to be a bit more human friendly. In case there is ever a need to debug the entries in the database it is easier to visually parse the key. I have no strong preference either way,  so I can change it to your suggestion if performance/complexity is an issue.

Reply:
readability is important you can keep it that way. we don't need to disassemble also so its mostly storing :)